.github/workflows/process_new_pr.yml

```diff
@@ -0,0 +1,127 @@
+name: Assign Reviewers for Contributors
+
+on:
+  pull_request_target:
+    types: [opened]
+
+permissions: {}
+
+jobs:
+  check_contributor_and_reviewers:
+    name: Check Contributor Status and Assigned Reviewers
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: read
+    outputs:
+      is_new_contributor: ${{ steps.check-new-contributor.outputs.result }}
+      has_reviewers: ${{ steps.check-reviewers.outputs.has_reviewers }}
+      has_teams: ${{ steps.check-reviewers.outputs.has_teams }}
+      author: ${{ steps.set-env.outputs.author }}
+      repo: ${{ steps.set-env.outputs.repo }}
+      owner: ${{ steps.set-env.outputs.owner }}
+      pr_number: ${{ steps.set-env.outputs.pr_number }}
+    steps:
+      - name: Extract PR Metadata
+        id: set-env
+        shell: bash
+        run: |
+          echo "author=${{ github.event.pull_request.user.login }}" >> $GITHUB_OUTPUT
+          echo "repo=${{ github.event.repository.name }}" >> $GITHUB_OUTPUT
+          echo "owner=${{ github.event.repository.owner.login }}" >> $GITHUB_OUTPUT
+          echo "pr_number=${{ github.event.pull_request.number }}" >> $GITHUB_OUTPUT
+      - name: Check if PR Author is a New Contributor
+        id: check-new-contributor
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
The workflow token is available by default, so you don't need to pass it separately. See https://github.com/actions/github-script?tab=readme-ov-file#examples.

ditto below